great_expectations/checkpoint/checkpoint.py

```diff
@@ -82,105 +101,216 @@ def __init__(
 
         self._data_context = data_context
 
-        checkpoint_config: CheckpointConfig = CheckpointConfig(
-            **{
-                "name": name,
-                "config_version": config_version,
-                "template_name": template_name,
-                "module_name": module_name,
-                "class_name": class_name,
-                "run_name_template": run_name_template,
-                "expectation_suite_name": expectation_suite_name,
-                "expectation_suite_ge_cloud_id": expectation_suite_ge_cloud_id,
-                "batch_request": batch_request,
-                "action_list": action_list,
-                "evaluation_parameters": evaluation_parameters,
-                "runtime_configuration": runtime_configuration,
-                "validations": validations,
-                "profilers": profilers,
-                "ge_cloud_id": ge_cloud_id,
-                # Next two fields are for LegacyCheckpoint configuration
-                "validation_operator_name": validation_operator_name,
-                "batches": batches,
-            }
+        self._action_list = action_list
+
+        self._validations = validations
+
+        self._ge_cloud_id = ge_cloud_id
+
+        config_kwargs: dict = {
+            "name": name,
+            "config_version": config_version,
+            "template_name": template_name,
+            "run_name_template": run_name_template,
+            "expectation_suite_name": expectation_suite_name,
+            "expectation_suite_ge_cloud_id": expectation_suite_ge_cloud_id,
+            "batch_request": batch_request or {},
+            "action_list": action_list or [],
+            "evaluation_parameters": evaluation_parameters or {},
+            "runtime_configuration": runtime_configuration or {},
+            "profilers": profilers or [],
+            "validations": validations or [],
+            "ge_cloud_id": ge_cloud_id,
+            # Next two fields are for LegacyCheckpoint configuration
+            "validation_operator_name": validation_operator_name,
+            "batches": batches,
+        }
+        self._config_kwargs = config_kwargs
```

Comment:
This is not the place for it but I think we should eventually type this as a dataclass or TypedDict

Reply:
@cdkini I added the `Attribute` class -- provides the dict/dot notation.  We can improve on it later, if desired.